### PR TITLE
fix: internal audit balanced issues fixed

### PR DESCRIPTION
**Pull request by Itshyphen**
This reverts commit 4aa2cc2aa01122fb5f2ef8a150914bc58e2df103, reversing changes made to 0ba36419405d6e5b84affc14660f7fa3f7b0e558.

I accidentally merged previous pr and reverted that, so created new one

### CODE REVIEW

**Review by AntonAndell — APPROVED** [review]
LGTM

File: sources/asset_manager.move [hunk 1]
```diff
@@ -298,11 +309,9 @@ module balanced::asset_manager{
...
                 ctx
             );
-            xcall::execute_call_result(xcall,ticket,true,fee,ctx);
-        }else{
-            xcall::execute_call_result(xcall,ticket,false,fee,ctx);
         };
-        
+
+        xcall::execute_call_result(xcall,ticket,true,fee,ctx);
     }
```
Comment (ibrizsabin) [inline sources/asset_manager.move:315]: we have to revert when type send in message and type in type arguments do not match  , but in these cases 
`   assert!(
            verified,
            ProtocolMismatch
        );

        let method: vector<u8> = deposit::get_method(&msg);
        assert!(
            method == WITHDRAW_TO_NAME || method == WITHDRAW_NATIVE_TO_NAME,
            UnknownMessageType
        );
        `   we can actually trigger rollback.

File: sources/asset_manager.move [hunk 2]
```diff
@@ -149,6 +148,22 @@ module balanced::asset_manager{
         config.assets.add(token_type, asset_manager)
     }
 
+    entry fun configure_rate_limit<T> (
+        _: &AdminCap,
+        config: &mut Config,
+        c: &Clock,
```
Comment (ibrizsabin) [inline sources/asset_manager.move:154]: Config param should be first by convention, then other params

**Review by ibrizsabin — COMMENTED** [review]
few comment

File: sources/asset_manager.move (hunk 1 above)
Comment (AntonAndell) [reply sources/asset_manager.move:315]: Can probably be the same for all contracts, revert if method does not exists or in case of protocol mismatch

**Pull request merged**